docs/connector-development/config-based/connector-builder-ui.md

```diff
@@ -0,0 +1,81 @@
+
+
+<!-- TODO remove this when we bundle schema specification into the YAML -->
+:::info
+Remember that you need to declare the schemas of your streams separately from the YAML manifest. 
+
+You can do this by placing one `.json` file per stream in the `schemas/` directory of your connector module. Each `.json` should be a JSONSchema describing the schema of your stream. More information about this can be found in the [static schemas section of the Python CDK](../cdk-python/schemas#static-schemas). 
+:::
+
+### Building the connector image
+Follow the instructions in the connector README to build the Docker image. Typically this will be something like `docker build . -t airbyte/source-<name>:<version>`. 
+
+Once you've built the connector image, [follow these instructions](../../integrations/custom-connectors#adding-your-connectors-in-the-ui) to add your connector to your Airbyte instance. 
+
+## Connector Builder Components
+
+The UI contains two main components: the text editor where you can edit your YAML (left) according to the [YAML lowcode specification](../understanding-the-yaml-file/yaml-overview), and the testing panel (right) where you can get feedback on how your connector works as a result of your YAML manifest.
+
+![Annotated Test Pane](./assets/lowcode_annotated_test_pane.png)
+
+1. **Input parameters panel**: Configure the input parameters to be used in testing. For example, if the connector requires an API key, use this panel to specify a JSON object containing your API key e.g: `{"api_key": "xyz"}`
+2. **Stream Picker dropdown**: Use this dropdown to choose which stream you want to test
+3. **Endpoint URL**: Displays the URL queried by the CDK to retrieve data for the current stream
+4. **Test button**: When clicked, retrieves the data for the selected stream using the stream configuration setup in the YAML file. This is the equivalent of running the `read` command on the terminal for a single stream (the selected stream). 
+5. **Records tab**: Displays the final output returned by the connector for the selected page of data in this stream
+6. **Request tab**: Displays the outgoing HTTP request made by the connector to retrieve the selected page of data. Useful for debugging. 
+7. **Response tab**: Displays the full HTTP response received by the connector for the selected page of data. Useful for debugging. 
+8. **Results view**: Displays information based on the selected tab
+9. **Page selector** Displays the selected page
+10. **Logs view**: Displays the logs emitted by the connector while running
+
+The following demo video demonstrates these components on a very simple API: 
+
+<div style={{position: "relative", "padding-bottom": "64.90384615384616%", height: 0}}>
+<iframe src="https://www.loom.com/embed/acf899938ef74dec8dd61ba012bc872f" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen style={{position: "absolute", top: 0, left: 0, width: "100%", height: "100%"}}></iframe>
```

Comment:
Something you noted in the video was that there were no connector logs returned for the poke API call. This is something I noticed also while testing -- it seems like the connector builder server doesn't really ever return logs. 

Is this just because the only LOG messages the CDK produces are those containing the HTTP request/response, and those are already being shuffled into other parts of the response? FYI @girarda @brianjlai

Reply:
it (the CDK)  definitely logs stuff when there's slices and whatnot

Reply:
Hmm I just tried out running the exchange_rates_api low-code connector (the one from the tutorial) in the connector builder UI, which has a stream slicer configured, and there still isn't any logs:
<img width="1333" alt="Screen Shot 2022-11-18 at 10 28 53 AM" src="https://user-images.githubusercontent.com/22731524/202777034-97bbac64-f0d1-4b81-9335-56efa4be2c7c.png">

I did see `DEBUG` messages being printed out by the server, so maybe the issue is that those `DEBUG` messages are not being captured/returned by the server?

Reply:
That's a good point. The only log messages being returned right now are the request and response.

[The CDK does log some messages through the logger though](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L235). 

[Created an issue to return AirbyteLogMessage instead (or in addition to using the logger)](https://github.com/airbytehq/airbyte/issues/19613)